pkg/assembler/backends/inmem/pkg.go

```diff
@@ -744,6 +744,21 @@ func noMatchQualifiers(filter *model.PkgSpec, v map[string]string) bool {
 }
 
 func (c *demoClient) exactPackageVersion(filter *model.PkgSpec) (*pkgVersionNode, error) {
+	pkgs, err := c.findPackageVersion(filter)
+	if err != nil {
+		return nil, gqlerror.Errorf("No package version matches spec %v", err)
+	}
+	switch len(pkgs) {
+	case 0:
+		return nil, nil
+	case 1:
+		return pkgs[0], nil
+	default:
+		return nil, gqlerror.Errorf("More than one package version matches spec")
```

Comment:
Thanks for doing this @mrizzi, so this would be a valid return. In that case, it should return all found and let the verb filters figure out which ones are valid per the query.

Reply:
hmm I think it should always be `findPackageVersion` as the `exactPackageVersion` is not valid. Meaning we need to search all returned matching pkgVersions backlinks for matching verbs.

Reply:
else we would have a similar issue in other verbs as we did with `pkgEquals`

Reply:
> hmm I think it should always be `findPackageVersion` as the `exactPackageVersion` is not valid. Meaning we need to search all returned matching pkgVersions backlinks for matching verbs.

Done :wink: